File: backend/app/schemas/folder.py
import uuid
from datetime import datetime

from pydantic import BaseModel, ConfigDict, Field


class FolderCreate(BaseModel):
    user_id: uuid.UUID
    name: str = Field(min_length=1, max_length=100)
    color: str = Field(default="violet", max_length=20)


class FolderUpdate(BaseModel):
    name: str | None = Field(default=None, min_length=1, max_length=100)
    color: str | None = Field(default=None, max_length=20)


class FolderRead(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: uuid.UUID
    user_id: uuid.UUID
    name: str
    color: str
    created_at: datetime
    piece_count: int = 0

File: frontend/src/types/folder.ts
export interface Folder {
  id: string;
  user_id: string;
  name: string;
  color: string;
  created_at: string;
  piece_count: number;
}

export const FOLDER_COLORS = ["violet", "emerald", "amber", "rose", "sky"] as const;
export type FolderColor = (typeof FOLDER_COLORS)[number];
